src/mod_private.erl

```diff
@@ -69,16 +69,19 @@ get_personal_data(Acc, HostType, #jid{ luser = LUser, lserver = LServer }) ->
 %% ------------------------------------------------------------------
 %% gen_mod callbacks
 
+-spec start(HostType :: mongooseim:host_type(), Opts :: gen_mod:module_opts()) -> ok.
 start(HostType, Opts) ->
-    mod_private_backend:init(HostType, Opts),
     IQDisc = gen_mod:get_opt(iqdisc, Opts, one_queue),
+    mod_private_backend:init(HostType, Opts),
     ejabberd_hooks:add(hooks(HostType)),
     gen_iq_handler:add_iq_handler_for_domain(HostType, ?NS_PRIVATE, ejabberd_sm,
                                              fun ?MODULE:process_iq/5, #{}, IQDisc).
 
+-spec stop(HostType :: mongooseim:host_type()) -> ok.
 stop(HostType) ->
     ejabberd_hooks:delete(hooks(HostType)),
-    gen_iq_handler:remove_iq_handler_for_domain(HostType, ?NS_PRIVATE, ejabberd_sm).
+    gen_iq_handler:remove_iq_handler_for_domain(HostType, ?NS_PRIVATE, ejabberd_sm),
+    ok.
```

Comment:
I'd rather leave the result of `remove_iq_handler_for_domain` and change the spec to `-> ok | {error, not_registered}`. gen_mod's stop callback can return `any()` from what I see. Is there a reason to return a `ok` here?

Reply:
It's not like it's a big issue, as the result is discarded anyway, but still...